MmSupervisorPkg/Drivers/MmPeiLaunchers/MmDxeSupport.c

```diff
@@ -709,6 +712,276 @@ SmmIplSetVirtualAddressNotify (
+      TempCommBuffer.NewCommBuffers[CommRegionHob->MmCommonRegionType].MemoryDescriptor.PhysicalStart = CommRegionHob->MmCommonRegionAddr;
+
+      // MU_CHANGE Starts: MM_SUPV: Fetch allocated communication buffer from HOBs
+      //                   And publish notification when the table is installed.
+      GuidHob.Guid = GET_NEXT_HOB (GuidHob);
+      GuidHob.Guid = GetNextGuidHob (&gMmCommonRegionHobGuid, GuidHob.Guid);
+    }
+  }
+
+  for (Index = 0; Index < MM_OPEN_BUFFER_CNT; Index++) {
+    PiSmmCommunicationRegionTable = NULL;
+    if (Index == MM_USER_BUFFER_T) {
+      ConfTableGuid = &gEdkiiPiSmmCommunicationRegionTableGuid;
+    } else if (Index == MM_SUPERVISOR_BUFFER_T) {
+      ConfTableGuid = &gMmSupervisorCommunicationRegionTableGuid;
+    }
+
+    DescriptorSize = sizeof (EFI_MEMORY_DESCRIPTOR);
+    //
+    // Make sure Size != sizeof(EFI_MEMORY_DESCRIPTOR). This will
+    // prevent people from having pointer math bugs in their code.
+    // now you have to use *DescriptorSize to make things work.
+    //
+    DescriptorSize += sizeof (UINT64) - (DescriptorSize % sizeof (UINT64));
+
+    //
+    // Allocate and fill PiSmmCommunicationRegionTable
+    //
+    PiSmmCommunicationRegionTable = AllocateReservedPool (sizeof (EDKII_PI_SMM_COMMUNICATION_REGION_TABLE) + DescriptorSize);
+    ASSERT (PiSmmCommunicationRegionTable != NULL);
+    // MU_CHANGE: MM_SUPV: Exit loop if allocation failed.
+    if (PiSmmCommunicationRegionTable == NULL) {
+      DEBUG ((DEBUG_ERROR, "%a Failed to allocate buffer for communication buffer table!!!\n", __FUNCTION__));
+      Status = EFI_OUT_OF_RESOURCES;
+      goto Done;
+    }
+
+    ZeroMem (PiSmmCommunicationRegionTable, sizeof (EDKII_PI_SMM_COMMUNICATION_REGION_TABLE) + DescriptorSize);
+
+    PiSmmCommunicationRegionTable->Version         = EDKII_PI_SMM_COMMUNICATION_REGION_TABLE_VERSION;
+    PiSmmCommunicationRegionTable->NumberOfEntries = 1;
+    PiSmmCommunicationRegionTable->DescriptorSize  = DescriptorSize;
+    Entry                                          = (EFI_MEMORY_DESCRIPTOR *)(PiSmmCommunicationRegionTable + 1);
+    Entry->Type                                    = EfiConventionalMemory;
+    Entry->PhysicalStart                           = (EFI_PHYSICAL_ADDRESS)(UINTN)TempCommBuffer.NewCommBuffers[Index].MemoryDescriptor.PhysicalStart; // MU_CHANGE: MM_SUPV: BAR from HOB
+    ASSERT (Entry->PhysicalStart != 0);
+    // MU_CHANGE: MM_SUPV: Exit loop if HOB data is null pointer.
+    if (Entry->PhysicalStart == 0) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "%a Target HOB does not contain valid communication buffer data: type: 0x%x, addr: 0x%p, size: 0x%x!!!\n",
+        __FUNCTION__,
+        Index,
+        TempCommBuffer.NewCommBuffers[Index].MemoryDescriptor.PhysicalStart,
+        TempCommBuffer.NewCommBuffers[Index].MemoryDescriptor.NumberOfPages
+        ));

```

Comment:
I think thi is what was intended.

```suggestion
      DEBUG ((
        DEBUG_ERROR,
        "%a Target HOB does not contain valid communication buffer data: type: 0x%x, addr: 0x%p, size: 0x%x!!!\n",
        __FUNCTION__,
        TempCommBuffer.NewCommBuffers[Index].MemoryDescriptor.Type,
        TempCommBuffer.NewCommBuffers[Index].MemoryDescriptor.PhysicalStart,
        EFI_PAGES_TO_SIZE (TempCommBuffer.NewCommBuffers[Index].MemoryDescriptor.NumberOfPages)
        ));
```

Reply:
For the type, it is really expecting the index (being buffer for user, or supervisor). but I updated the print to `pages` to reduce unnecessary computation.